spiped/dispatch.c

```diff
@@ -94,13 +103,36 @@ doaccept(struct accept_state * A)
 
 /* A connection has closed.  Accept more if necessary. */
 static int
-callback_conndied(void * cookie)
+callback_conndied(void * cookie, void * conn_cookie)
 {
 	struct accept_state * A = cookie;
+	struct conn_list_node * node_ptr;
+
+	assert(A->conn_cookies != NULL);
 
 	/* We've lost a connection. */
 	A->nconn -= 1;
 
+	/* Find the node that contains conn_cookie. */
```

Comment:
My question still remains.  You're walking a linked list looking for something here; walking linked lists looking for something is slow.